core/src/main/java/com/linecorp/armeria/server/ProxiedAddresses.java

```diff
@@ -18,51 +18,86 @@
-    private final SocketAddress destinationAddress;
+    /**
+     * Creates a new instance with the specified {@code sourceAddress} and {@code destinationAddress}.
+     */
+    public static ProxiedAddresses of(InetSocketAddress sourceAddress, InetSocketAddress destinationAddress) {
+        return new ProxiedAddresses(sourceAddress, ImmutableList.of(destinationAddress));
+    }
+
+    /**
+     * Creates a new instance with the specified {@code sourceAddress} and {@code destinationAddresses}.
+     */
+    public static ProxiedAddresses of(InetSocketAddress sourceAddress,
+                                      Iterable<? extends InetSocketAddress> destinationAddresses) {
+        return new ProxiedAddresses(sourceAddress, ImmutableList.copyOf(destinationAddresses));
+    }
+
+    private final InetSocketAddress sourceAddress;
+    private final List<InetSocketAddress> destinationAddresses;
 
-    private ProxiedAddresses(SocketAddress sourceAddress, SocketAddress destinationAddress) {
+    private ProxiedAddresses(InetSocketAddress sourceAddress, List<InetSocketAddress> destinationAddresses) {
         this.sourceAddress = requireNonNull(sourceAddress, "sourceAddress");
-        this.destinationAddress = requireNonNull(destinationAddress, "destinationAddress");
+        this.destinationAddresses = requireNonNull(destinationAddresses, "destinationAddresses");
     }
 
     /**
      * Returns the source address of the proxied request.
      */
-    @SuppressWarnings("unchecked")
-    public <A extends SocketAddress> A sourceAddress() {
-        return (A) sourceAddress;
+    public InetSocketAddress sourceAddress() {
+        return sourceAddress;
     }
 
     /**
-     * Returns the destination address of the proxied request.
+     * Returns the destination addresses of the proxied request.
      */
-    @SuppressWarnings("unchecked")
-    public <A extends SocketAddress> A destinationAddress() {
-        return (A) destinationAddress;
+    public List<InetSocketAddress> destinationAddresses() {
+        return destinationAddresses;
     }
 
     @Override
     public String toString() {
         return MoreObjects.toStringHelper(this)
                           .add("sourceAddress", sourceAddress)
-                          .add("destinationAddress", destinationAddress)
+                          .add("destinationAddress", destinationAddresses)
                           .toString();
     }
+
+    @Override
+    public boolean equals(final Object o) {
+        if (this == o) {
+            return true;
+        }
+
+        if (!(o instanceof ProxiedAddresses)) {
+            return false;
+        }
+
+        final ProxiedAddresses that = (ProxiedAddresses) o;
+        return sourceAddress.equals(that.sourceAddress) &&
+               destinationAddresses.equals(that.destinationAddresses);
+    }
+
+    @Override
+    public int hashCode() {
+        return sourceAddress.hashCode() * 31 + destinationAddresses.hashCode();
+    }
```

Comment:
`toString()` should come before `hashCode` and `equals`, like other classes.

Reply:
You mean `toString()` should come after ...? 😄

Reply:
YEEES :rofl

Reply:
I will keep in mind. 😎